util/chunk/chunk.go

```diff
@@ -39,11 +43,41 @@ const (
 
 // NewChunkWithCapacity creates a new chunk with field types and capacity.
 func NewChunkWithCapacity(fields []*types.FieldType, cap int) *Chunk {
+	return New(fields, cap, cap) //FIX ME in continue PR, this keep every thing same.
+}
+
+// New creates a new chunk.
+//  cap: the max number of rows that the chunk want to hold.
+//  capLimit: the max limit for max number of rows.
+func New(fields []*types.FieldType, cap, capLimit int) *Chunk {
 	chk := new(Chunk)
 	chk.columns = make([]*column, 0, len(fields))
 	chk.numVirtualRows = 0
 	for _, f := range fields {
-		chk.addColumnByFieldType(f, cap)
+		chk.addColumnByFieldType(getFixedLen(f), cap, nil)
+	}
+	chk.capacity = cap
+	if chk.capacity > capLimit {
+		chk.capacity = capLimit
+	}
+	return chk
+}
+
+// Renew recreates a new chunk base on old chunk.
+//   chk: old chunk(often used in previous call).
+//   capLimit: the max limit for max number of rows.
+func Renew(chk *Chunk, capLimit int) *Chunk {
```

Comment:
When do we need this function?

Reply:
just see test case in situation that we want to renew instead of resue.